src/libsyntax/feature_gate.rs

```diff
@@ -1220,72 +1221,114 @@ pub const BUILTIN_ATTRIBUTES: &[(&str, AttributeType, AttributeTemplate, Attribu
-                                 "rustc_attrs",
-                                 "never will be stable",
-                                 cfg_fn!(rustc_attrs))),
+    (rustc_args_required_const, Whitelisted, template!(List: "N"), Gated(Stability::Unstable,
+                                "rustc_attrs",
+                                "never will be stable",
+                                cfg_fn!(rustc_attrs))),
     // RFC 2070
-    ("panic_handler", Normal, template!(Word), Ungated),
+    (panic_handler, Normal, template!(Word), Ungated),
 
-    ("alloc_error_handler", Normal, template!(Word), Gated(Stability::Unstable,
-                           "alloc_error_handler",
-                           "#[alloc_error_handler] is an unstable feature",
-                           cfg_fn!(alloc_error_handler))),
+    (alloc_error_handler, Normal, template!(Word), Gated(Stability::Unstable,
+                        "alloc_error_handler",
+                        "#[alloc_error_handler] is an unstable feature",
+                        cfg_fn!(alloc_error_handler))),
 
     // RFC 2412
-    ("optimize", Whitelisted, template!(List: "size|speed"), Gated(Stability::Unstable,
-                               "optimize_attribute",
-                               "#[optimize] attribute is an unstable feature",
-                               cfg_fn!(optimize_attribute))),
+    (optimize, Whitelisted, template!(List: "size|speed"), Gated(Stability::Unstable,
+                            "optimize_attribute",
+                            "#[optimize] attribute is an unstable feature",
+                            cfg_fn!(optimize_attribute))),
 
     // Crate level attributes
-    ("crate_name", CrateLevel, template!(NameValueStr: "name"), Ungated),
-    ("crate_type", CrateLevel, template!(NameValueStr: "bin|lib|..."), Ungated),
-    ("crate_id", CrateLevel, template!(NameValueStr: "ignored"), Ungated),
-    ("feature", CrateLevel, template!(List: "name1, name1, ..."), Ungated),
-    ("no_start", CrateLevel, template!(Word), Ungated),
-    ("no_main", CrateLevel, template!(Word), Ungated),
-    ("no_builtins", CrateLevel, template!(Word), Ungated),
-    ("recursion_limit", CrateLevel, template!(NameValueStr: "N"), Ungated),
-    ("type_length_limit", CrateLevel, template!(NameValueStr: "N"), Ungated),
-    ("test_runner", CrateLevel, template!(List: "path"), Gated(Stability::Unstable,
+    (crate_name, CrateLevel, template!(NameValueStr: "name"), Ungated),
+    (crate_type, CrateLevel, template!(NameValueStr: "bin|lib|..."), Ungated),
+    (crate_id, CrateLevel, template!(NameValueStr: "ignored"), Ungated),
+    (feature, CrateLevel, template!(List: "name1, name1, ..."), Ungated),
+    (no_start, CrateLevel, template!(Word), Ungated),
+    (no_main, CrateLevel, template!(Word), Ungated),
+    // FIXME: Duplicated
+    //(no_builtins, CrateLevel, template!(Word), Ungated),
```

Comment:
Yeah, could you remove [the second instances of `no_builtins` and `no_mangle`](https://github.com/rust-lang/rust/pull/57321#issuecomment-467681091), they never had any effect.

Reply:
Done.